cmd/kava/opendb/props_loader.go

```diff
@@ -0,0 +1,87 @@
+//go:build rocksdb
+// +build rocksdb
+
+package opendb
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/pkg/errors"
+)
+
+type propsGetter interface {
+	GetProperty(propName string) (value string)
+	GetIntProperty(propName string) (value uint64, success bool)
+}
+
+type propsLoader struct {
+	db        propsGetter
+	errorMsgs []string
+}
+
+func newPropsLoader(db propsGetter) *propsLoader {
+	return &propsLoader{
+		db:        db,
+		errorMsgs: make([]string, 0),
+	}
+}
+
+func (l *propsLoader) load() (*properties, error) {
+	props := &properties{
+		BaseLevel:               l.getIntProperty("rocksdb.base-level"),
+		BlockCacheCapacity:      l.getIntProperty("rocksdb.block-cache-capacity"),
+		BlockCachePinnedUsage:   l.getIntProperty("rocksdb.block-cache-pinned-usage"),
+		BlockCacheUsage:         l.getIntProperty("rocksdb.block-cache-usage"),
+		CurSizeActiveMemTable:   l.getIntProperty("rocksdb.cur-size-active-mem-table"),
+		CurSizeAllMemTables:     l.getIntProperty("rocksdb.cur-size-all-mem-tables"),
+		EstimateLiveDataSize:    l.getIntProperty("rocksdb.estimate-live-data-size"),
+		EstimateNumKeys:         l.getIntProperty("rocksdb.estimate-num-keys"),
+		EstimateTableReadersMem: l.getIntProperty("rocksdb.estimate-table-readers-mem"),
+		LiveSSTFilesSize:        l.getIntProperty("rocksdb.live-sst-files-size"),
+		SizeAllMemTables:        l.getIntProperty("rocksdb.size-all-mem-tables"),
+		OptionsStatistics:       l.getProperty("rocksdb.options-statistics"),
+	}
+
+	if len(l.errorMsgs) != 0 {
+		errorMsg := strings.Join(l.errorMsgs, ";")
+		return nil, errors.New(errorMsg)
+	}
+
+	return props, nil
+}
+
+func (l *propsLoader) getProperty(propName string) string {
+	value := l.db.GetProperty(propName)
+	if value == "" {
+		l.errorMsgs = append(l.errorMsgs, fmt.Sprintf("property %v is empty", propName))
+		return ""
```

Comment:
This is c style coding, go idiom is to return `value, error` and have callers check if err is nil before using value

Reply:
https://go.dev/blog/error-handling-and-go

Reply:
generally I agree, but I believe in some cases such approach make sense to allow code like this:
https://github.com/Kava-Labs/kava/blob/yevhenii/rocksdb-metrics/cmd/kava/opendb/stats_loader.go#L157-L193

otherwise instead of `40 lines of code`, I'll have `120 lines of code` (most of them will be error-checking)

also I put some comments regarding this:
`// NOTE: some methods accumulate errors instead of returning them, these methods are private and not intended to use outside`